webdriver/tests/bidi/browsing_context/set_viewport/set_viewport.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from ... import get_viewport_dimensions
+
+@pytest.mark.asyncio
+async def test_set_viewport(bidi_session, new_tab):
+    test_viewport = {"width": 250, "height": 300}
+
+    await bidi_session.browsing_context.set_viewport(
+        context=new_tab["context"],
+        viewport=test_viewport)
+    
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_reset(bidi_session, new_tab):
+    original_viewport = await get_viewport_dimensions(bidi_session, new_tab)
+
+    test_viewport = {"width": 333, "height": 333}
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=None)
+    assert await get_viewport_dimensions(bidi_session, new_tab) == original_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_affects_specific_context(bidi_session, top_context, new_tab):
+    original_viewport = await get_viewport_dimensions(bidi_session, top_context)
+
+    test_viewport = {"width": 333, "height": 333}
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+    assert await get_viewport_dimensions(bidi_session, top_context) == original_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_persists_on_navigation(bidi_session, new_tab, inline):
```

Comment:
Would you consider to parameterize this method as suggested so that we also have a coop navigation? Here the existing browsing context will definitely be replace which will not happen for a navigation like the one we have right now.

Reply:
yeah, I am still looking into that